user_guide.adoc

```diff
@@ -1377,30 +1377,82 @@ library" for expression language.
 |Return type
 |Description
 
-|to_s
+|`to_s`
 |String
 |Converts integer into a string using decimal representation
-
-|to_s(radix)
-|String
-|Converts integer into a string using specified `radix` (i.e. use `16` to get hexadecimal representation, use `8` to get octal, etc)
 |===
 
 ==== Floating point numbers
 
-TODO
+No methods are currently supported for floating point numbers.
 
 ==== Byte arrays
 
-TODO
+[cols="3*", options="header"]
+|===
+|Method name
+|Return type
+|Description
+
+|`to_s(encoding)`
+|String
+|Decodes (converts) a byte array encoded using the specified `encoding` scheme into a string
+|===
 
 ==== Strings
 
-TODO
+[cols="3*", options="header"]
+|===
+|Method name
+|Return type
+|Description
+
+|`length`
+|Integer
+|Length of a string in number of bytes
```

Comment:
Generally, *a lot* of languages do not care and have a single "integer" type, which is signed and either platform-dependent or infinite (transparently converting to big integer or something like that as it grows). To accomodate that, we just take it as common denominator and state that all returned integers are `CalcIntType`, and they are mapped to some common one-size-fits-most integer type in a target language. And for many languages, it's the only type ;)

Actually, even in languages that do care about many integer types, there's still some degree of int coalescing. For example, if you add `byte` and `byte` in Java, you get `int` automatically. I'm not sure about C++, but I totally won't be surprised if you end up with `int` when trying to add `uint8_t` and `int16_t`, for example.